### ISSUE TITLE
Laravel seeRecord expects empty row in mysql to be compared to null instead of empty

### ISSUE THREAD

**Issue by dakipro** [Codeception/Codeception#4897]
#### What are you trying to achieve?
I have items that might or might not have a name set. If it is set to some string test works fine, but if it is empty string, codeception doesn't find it in database when comparing it to the model field. 
This works when $item->name is set to f.eks. 'abc' but not if item name is set to ''
$I->seeRecord(Item::class, [
            'id' => $lastId
            'name' => $item->name
]);

It is not a show stopper, but doesn't seem logical to me that it would missmatch the type (or am I doing something wrong?)

#### What do you get instead?
fail that Could not find App\Models\Item\Item with {"id":"1192","name":""}
I then have to compare it with "null" and it works fine. So this would work fine
 'name' => $item->name == ''? null : $item->name,

I have checked in database, and it is an empty string, not a null value

> Provide console output if related. Use `-vvv` mode for more details.

```bash
# paste output here
```
> Provide test source code if related

```php
 $item = factory(Item::class)->make([
            'name' => '',  
        ]);

        // Act
        $I->sendPOST('items/', $item);

        // Assert
        $lastId = DB::getPdo()->lastInsertId();

        $I->seeRecord(Item::class, [
            'id' => $lastId,
            'name' => $item->name,   // this doesn't find the record
        ]);

/* 
- if I set 'name' to f.eks. 'abc' in the factory code above
    then seeRecord code from above would find the record just fine. 

- this works fine 
 $I->seeRecord(Item::class, [
            'id' => $lastId,
            'name' => $item->name == ''? null : $item->name,
        ]);

*/

```
### Details

* Codeception version: 2.4.0
* PHP Version: 7.1.9
* Operating System: Windows
* Installation type: Composer
* List of installed packages (`composer show`)
arcanedev/log-viewer                  4.5.1
arcanedev/support                     4.3.0
asm89/stack-cors                      1.2.0
barryvdh/laravel-cors                 v0.11.0
barryvdh/laravel-debugbar             v3.1.3
barryvdh/laravel-ide-helper           v2.4.3
barryvdh/reflection-docblock          v2.0.4
behat/gherkin                         v4.5.1
codeception/codeception               2.4.0
codeception/phpunit-wrapper           7.0.5
codeception/stub                      1.0.2
creativeorange/gravatar               1.0.10
darkaonline/l5-swagger                5.6.2
davejamesmiller/laravel-breadcrumbs   5.0.0
defuse/php-encryption                 v2.1.0
dnoegel/php-xdg-base-dir              0.1
doctrine/annotations                  v1.6.0
doctrine/cache                        v1.7.1
doctrine/collections                  v1.5.0
doctrine/common                       v2.8.1
doctrine/dbal                         v2.6.3
doctrine/inflector                    v1.3.0
doctrine/instantiator                 1.1.0
doctrine/lexer                        v1.0.1
dragonmantank/cron-expression         v2.0.0
egulias/email-validator               2.1.3
erusev/parsedown                      1.7.0
facebook/webdriver                    1.5.0
fideloper/proxy                       4.0.0
filp/whoops                           2.1.14
firebase/php-jwt                      v5.0.0
flow/jsonpath                         0.4.0
fzaninotto/faker                      v1.7.1
guzzlehttp/guzzle                     6.3.0
guzzlehttp/promises                   v1.3.1
guzzlehttp/psr7                       1.4.2
hamcrest/hamcrest-php                 v1.2.2
hieu-le/active                        3.5.1
jakub-onderka/php-console-color       0.1
jakub-onderka/php-console-highlighter v0.3.2
laravel/browser-kit-testing           dev-master 3ea22c4
laravel/framework                     v5.6.7
laravel/passport                      v5.0.3
laravel/tinker                        v1.0.3
laravelcollective/html                v5.6.3
lcobucci/jwt                          3.2.2
league/event                          2.1.2
league/flysystem                      1.0.42
league/oauth2-server                  6.1.1
maximebf/debugbar                     v1.15.0
mockery/mockery                       0.9.9
monolog/monolog                       1.23.0
myclabs/deep-copy                     1.7.0
nesbot/carbon                         1.23.0
nikic/php-parser                      v3.1.4
paragonie/random_compat               v2.0.11
phar-io/manifest                      1.0.1
phar-io/version                       1.0.1
phpdocumentor/reflection-common       1.0.1
phpdocumentor/reflection-docblock     4.3.0
phpdocumentor/type-resolver           0.4.0
phpseclib/phpseclib                   2.0.10
phpspec/prophecy                      1.7.5
phpunit/php-code-coverage             6.0.1
phpunit/php-file-iterator             1.4.5
phpunit/php-text-template             1.2.1
phpunit/php-timer                     2.0.0
phpunit/php-token-stream              3.0.0
phpunit/phpunit                       7.0.2
phpunit/phpunit-mock-objects          6.0.1
psr/container                         1.0.0
psr/http-message                      1.0.1
psr/log                               1.0.2
psr/simple-cache                      1.0.0
psy/psysh                             v0.8.17
ramsey/uuid                           3.7.3
sebastian/code-unit-reverse-lookup    1.0.1
sebastian/comparator                  2.1.3
sebastian/diff                        3.0.0
sebastian/environment                 3.1.0
sebastian/exporter                    3.1.0
sebastian/global-state                2.0.0
sebastian/object-enumerator           3.0.3
sebastian/object-reflector            1.1.1
sebastian/recursion-context           3.0.0
sebastian/resource-operations         1.0.0
sebastian/version                     2.0.1
swagger-api/swagger-ui                v3.11.0
swiftmailer/swiftmailer               v6.0.2
symfony/browser-kit                   v4.0.6
symfony/class-loader                  v3.4.4
symfony/console                       v4.0.4
symfony/css-selector                  v4.0.4
symfony/debug                         v4.0.4
symfony/dom-crawler                   v4.0.4
symfony/event-dispatcher              v4.0.4
symfony/finder                        v4.0.4
symfony/http-foundation               v4.0.4
symfony/http-kernel                   v4.0.4
symfony/polyfill-mbstring             v1.7.0
symfony/polyfill-php72                v1.7.0
symfony/process                       v4.0.4
symfony/psr-http-message-bridge       v1.0.2
symfony/routing                       v4.0.4
symfony/translation                   v4.0.4
symfony/var-dumper                    v4.0.4
symfony/yaml                          v4.0.6
theseer/tokenizer                     1.1.0
tijsverkoyen/css-to-inline-styles     2.2.1
vlucas/phpdotenv                      v2.4.0
webmozart/assert                      1.3.0
yajra/laravel-datatables-oracle       v8.4.1
zendframework/zend-diactoros          1.7.1
zircote/swagger-php                   2.0.13

* Suite configuration:

```yml
# actor: ApiTester
modules:
    enabled:
        - REST:
            url: http://api.mytotalcollection-laravel.dloc/api/v1/
            depends: Laravel5
        - \Helper\Api
        - Laravel5:
            environment_file: .env.functional.testing


```

**Comment by DavertMik** [comment]
Hi. Currently, we don't have a regular Laravel module maintainer, so If you have some time you can help us. Yes, you can start with a pull request fixing this issue. Thanks!

**Comment by dakipro** [comment]
Thanks, I will certainly try to contribute if I get the time. 
Regarding the issue, looks like it is very specific use case, as I cannot reproduce it anymore (I did some heavy refactoring of the code, and perhaps I lost exact point where I needed to use workaround). 
If I run into it again I will look more deeply into it

**Comment by Naktibalda** [comment]
Fixed by #5484
